base/compiler/ssair/inlining.jl

```diff
@@ -1321,26 +1418,30 @@ end
 
 function handle_match!(
     match::MethodMatch, argtypes::Vector{Any}, flag::UInt8, state::InliningState,
-    cases::Vector{InliningCase}, allow_abstract::Bool = false)
+    cases::Vector{InliningCase}, allow_abstract::Bool = false, check_sparams::Bool = false)
     spec_types = match.spec_types
     allow_abstract || isdispatchtuple(spec_types) || return false
-    # we may see duplicated dispatch signatures here when a signature gets widened
-    # during abstract interpretation: for the purpose of inlining, we can just skip
-    # processing this dispatch candidate
-    _any(case->case.sig === spec_types, cases) && return true
-    item = analyze_method!(match, argtypes, flag, state)
+    if check_sparams
+        # we may see duplicated dispatch signatures here when a signature gets widened
+        # during abstract interpretation: for the purpose of inlining, we can just skip
+        # processing this dispatch candidate
+        _any(case->case.sig === spec_types, cases) && return true
+        item = analyze_method!(match, argtypes, flag, state, true)
+    else
+        item = analyze_method!(match, argtypes, flag, state)
+    end
     item === nothing && return false
     push!(cases, InliningCase(spec_types, item))
     return true
 end
 
 function handle_const_prop_result!(
     result::ConstPropResult, argtypes::Vector{Any}, flag::UInt8, state::InliningState,
-    cases::Vector{InliningCase}, allow_abstract::Bool = false)
+    cases::Vector{InliningCase}, allow_abstract::Bool = false, check_sparams::Bool=false)
```

Comment:
There seems to be no callsites of this function with `check_sparams` on.

Reply:
I'd like to add on to this PR and allow unmatched type params for this kind of result in some cases, but for now I will remove this